Util/src/main/java/io/deephaven/util/codec/SimpleByteArrayCodec.java

```diff
@@ -0,0 +1,77 @@
+package io.deephaven.util.codec;
+
+import io.deephaven.datastructures.util.CollectionUtil;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class SimpleByteArrayCodec implements ObjectCodec<byte[]> {
```

Comment:
This warrants a top-level comment on what we're doing with this codec and what its guarantees are.